File: samples/python/routing/content_based_router.py
from langchain_anthropic import ChatAnthropic
from typing import Literal

llm = ChatAnthropic(model="claude-haiku-4-5-20251001")


def router_agent(task: str) -> Literal["coding", "research", "data"]:
    prompt = (
        f"Classify this task into exactly one category: coding, research, or data.\n"
        f"Task: {task}\n"
        f"Reply with only the single word: coding, research, or data."
    )
    result = llm.invoke(prompt)
    category = result.content.strip().lower()
    for valid in ("coding", "research", "data"):
        if valid in category:
            return valid  # type: ignore
    return "research"


def coding_agent(task: str) -> str:
    return llm.invoke(f"You are a coding expert. Answer briefly: {task}").content


def research_agent(task: str) -> str:
    return llm.invoke(f"You are a research expert. Answer briefly: {task}").content


def data_agent(task: str) -> str:
    return llm.invoke(f"You are a data analyst. Answer briefly: {task}").content


SPECIALISTS = {
    "coding": coding_agent,
    "research": research_agent,
    "data": data_agent,
}


def route(task: str) -> str:
    category = router_agent(task)
    print(f"  [router] classified as: {category}")
    specialist = SPECIALISTS[category]
    return specialist(task)


def main() -> None:
    tasks = [
        "Write a Python function to reverse a linked list.",
        "What were the main causes of the French Revolution?",
        "What is the average salary in the dataset if values are 50k, 75k, 90k, 65k?",
    ]

    for task in tasks:
        print(f"\nTask: {task}")
        answer = route(task)
        print(f"Answer: {answer}")


if __name__ == "__main__":
    main()

File: samples/typescript/routing/content_based_router.ts
import { ChatAnthropic } from "@langchain/anthropic";
const llm = new ChatAnthropic({ model: "claude-haiku-4-5-20251001" });
async function route(task: string) {
  const cat = (await llm.invoke(`Classify into one word: "coding", "research", or "data". Task: ${task}. Respond with one word only.`)).content as string;
  const label = cat.toLowerCase().includes("cod") ? "coding" : cat.toLowerCase().includes("res") ? "research" : "data";
  const specialist = await llm.invoke(`You are a ${label} specialist. Answer briefly: ${task}`);
  console.log(`[${label}] ${task.substring(0, 40)}: ${(specialist.content as string).substring(0, 80)}`);
}
async function main() {
  await route("How do I reverse a linked list in Python?");
  await route("What caused World War I?");
  await route("What is the average of [4, 8, 15, 16, 23, 42]?");
}
main().catch(console.error);
